sql/schemas.go

```diff
@@ -150,21 +150,32 @@ func NewPrimaryKeySchema(s Schema, pkOrds ...int) PrimaryKeySchema {
 
 // SchemaToPrimaryKeySchema adapts the schema given to a PrimaryKey schema using the primary keys of the table given, if
 // present. The resulting PrimaryKeySchema may have an empty key set if the table has no primary keys. Matching for
-// ordinals is performed by column name.
-func SchemaToPrimaryKeySchema(table Table, sch Schema) PrimaryKeySchema {
+// ordinals is performed by column name, with the aid of |renames| when provided.
+func SchemaToPrimaryKeySchema(table Table, sch Schema, renames ...ColumnRename) PrimaryKeySchema {
 	var pks []*Column
 	if pkt, ok := table.(PrimaryKeyTable); ok {
 		schema := pkt.PrimaryKeySchema()
 		for _, ordinal := range schema.PkOrdinals {
 			pks = append(pks, schema.Schema[ordinal])
 		}
+	} else {
+		// set PkOrdinals by schema order
+		return NewPrimaryKeySchema(sch)
+	}
+
+	mapping := make(map[string]string)
```

Comment:
`mapping` is a bit generic as a name. Maybe change to `oldToNewNames` or something along those lines.